### PR TITLE
feat: Flexibility to specify a config file without a separate config map

### PR DESCRIPTION
**Pull request by tiagoposse**
Add renovate.configFile and env keywords to values.

These changes allow injecting a config file via a init container (fetches the required token from an external secrets manager and injects a template into config.json) in a simple way that still keeps the requirement for config or configFile.

### CODE REVIEW

**Review by viceice — APPROVED** [review]

**Comment by JamieMagee** [comment]
Overall looks okay. Just needs the docs to be regenerated.

Can I ask about your use case: Is it not possible to store the entire config as an existing secret? Is there something about your configuration that needs to be frequently updated or generated at runtime? Just curious how people are using this chart.

**Comment by viceice** [comment]
@JamieMagee i think the major use case is to generate some short living access token, eg when using custom github app for renovate. 😉

**Review by viceice — CHANGES_REQUESTED** [review]
needs lint fix:

```diff
diff --git a/charts/renovate/README.md b/charts/renovate/README.md
index 828f507..3531ca7 100644
--- a/charts/renovate/README.md
+++ b/charts/renovate/README.md
@@ -47,6 +47,7 @@ The following table lists the configurable parameters of the chart and the defau
 | cronjob.labels | object | `{}` |  |
 | cronjob.schedule | string | `"0 1 * * *"` |  |
 | cronjob.successfulJobsHistoryLimit | string | `""` |  |
+| env | object | `{}` |  |
 | envFrom | list | `[]` |  |
 | existingSecret | string | `""` |  |
 | image.pullPolicy | string | `"IfNotPresent"` |  |
@@ -60,6 +61,7 @@ The following table lists the configurable parameters of the chart and the defau
 | redis.nameOverride | string | `"redis"` |  |
 | redis.usePassword | bool | `false` |  |
 | renovate.config | string | `""` |  |
+| renovate.configFile | string | `""` |  |
 | resources | object | `{}` |  |
 | secrets | object | `{}` |  |
 | serviceAccount.annotations | object | `{}` |  |
```

**Comment by tiagoposse** [comment]
There are two use cases, the first covered by @viceice, the second is related to storing the token in AWS SecretsManager/Vault/Google Secrets Manager and not giving developers access permission to it (it would allow them to impersonate another entity in gitlab), while still allowing those same devs to change some configs of renovate (e.g. the log level) without messing with the secret. Hope this clears it up :)

**Comment by viceice** [comment]
I see, it's currently not possible to use a `config.js` file, with that you could required additional dynamically generated files.

**Comment by tiagoposse** [comment]
I've realised there was a mistake I didn't test correctly, I pushed a change. the volume and volumeMounts section need to be adapted .Values.renovate.config is not passed. Considering so, maybe it's worth having a better name for configFile, that indicates this is supposed to be passed in a way completely external to the chart?

**Comment by JamieMagee** [comment]
What about `existingConfigFile`? It's in line with `existingSecret`.

**Comment by tiagoposse** [comment]
Updates done and the commits were squashed. If it looks good, I'm happy with the result and should be okay to merge.

**Comment by viceice** [comment]
@tiagoposse Please do not squash your commits, we will do a squash merge anyways, so your pr only needs to be up-to-date 😉

File: charts/renovate/README.md [hunk 1]
```diff
@@ -60,6 +61,7 @@ The following table lists the configurable parameters of the chart and the defau
 | redis.nameOverride | string | `"redis"` |  |
 | redis.usePassword | bool | `false` |  |
 | renovate.config | string | `""` |  |
+| renovate.externalConfigFile | string | `"Bring your own config file, using your own mount"` |  |
```
Comment (viceice) [inline charts/renovate/README.md:64]: Why `externalConfigFile` ? Didn't we suggest `existingConfigFile` ? 🤔

Comment (tiagoposse) [reply charts/renovate/README.md]: You did, my bad here. Corrected

File: charts/renovate/templates/configmap.yaml [hunk 2]
```diff
@@ -6,4 +7,5 @@ metadata:
     {{- include "renovate.labels" . | nindent 4 }}
 data:
   config.json: |-
-    {{ required "A valid .Values.renovate.config entry is required!" .Values.renovate.config | nindent 4 }}
\ No newline at end of file
+    {{ required "A valid .Values.renovate.config entry is required if existingConfigFile is not specified!" .Values.renovate.config | nindent 4 }}
+{{- end }}
```
Comment (viceice) [inline charts/renovate/templates/configmap.yaml]:
```suggestion
{{- end }}

```

newline at end

**Review by viceice — APPROVED** [review]
LGTM

@JamieMagee Can you please check this too

**Review by viceice — CHANGES_REQUESTED** [review]
needs some changes in readme:
```diff
diff --git a/charts/renovate/README.md b/charts/renovate/README.md
+ curl --silent --show-error --fail --location --output /tmp/helm-docs.tar.gz https://github.com/norwoodj/helm-docs/releases/download/v1.5.0/helm-docs_1.5.0_Linux_x86_64.tar.gz
index e29a8ec..a641f54 100644
+ tar -xf /tmp/helm-docs.tar.gz helm-docs
--- a/charts/renovate/README.md
+++ b/charts/renovate/README.md
@@ -61,7 +61,7 @@ The following table lists the configurable parameters of the chart and the defau
 | redis.nameOverride | string | `"redis"` |  |
+ ./helm-docs
time="2021-05-17T08:37:04Z" level=info msg="Found Chart directories [charts/renovate]"
 | redis.usePassword | bool | `false` |  |
 | renovate.config | string | `""` |  |
-| renovate.existingConfigFile | string | `"Bring your own config file, using your own mount"` |  |
+| renovate.existingConfigFile | string | `""` |  |
 | resources | object | `{}` |  |
 | secrets | object | `{}` |  |
 | serviceAccount.annotations | object | `{}` |  |
time="2021-05-17T08:37:04Z" level=info msg="Generating README Documentation for chart /home/runner/work/helm-charts/helm-charts/charts/renovate"
+ git diff --exit-code
```

**Review by viceice — APPROVED** [review]

**Pull request merged**